.eslintrc.js

```diff
@@ -75,6 +75,16 @@ module.exports = {
     '@typescript-eslint/ban-ts-comment': ['warn'],
     '@typescript-eslint/no-var-requires': ['warn'],
   },
+  overrides: [
+    {
+      files: ['*.js', '*.vue'],
+      extends: ['plugin:jsdoc/recommended'],
+      rules: {
+        'jsdoc/check-tag-names': ['warn', { definedTags: ['values'] }],
+        'jsdoc/require-hyphen-before-param-description': ['warn'],
```

Comment:
```suggestion
        'jsdoc/check-tag-names': ['error', { definedTags: ['values'] }],
        'jsdoc/require-hyphen-before-param-description': ['error'],
```

For something that is purely style I don't see the value in warnings at all. At most they'll just be annoying and still get ignored and we still won't have consistency (but every PR will be plagued by 20 additional ESLint comments added by the CI).

Reply:
I wanted these to be warnings till we fix these issues and then escalate them to errors.

Reply:
I guess I don't fully see what the plan is. Is it to merge this then open separate PRs to fix the various issues? Or a single follow up PR to fix them all? Why not just fix them in this PR, considering they're not runtime changes and present no risk to the app's functionality? It'd also give us a sense of the scope of the rules being proposed and whether they follow the existing style of the repository or if we'd be imposing a new style (either of which can be okay, it's just something we should considering and be aware of as we're merging in new linting rules).

Reply:
All the rules defined in `plugin:jsdoc/recommended` are also warnings by default.

Reply:
That's a shame, they don't even have a `recommended-error` or something :disappointed: 

Here would be my recommendation:

```json
    "rules": {
        "jsdoc/check-access": ['off'], // The `@access` tag isn't useful with modules
        "jsdoc/check-alignment": ['error'], // This prevents misaligned astriks
        "jsdoc/check-examples": ['off'], // This rule doesn't work with eslint 8
        "jsdoc/check-indentation": ['error'], // Prevents extra indentation outside of `@example`
        "jsdoc/check-line-alignment": ['error'], // I don't personally like this rule but Gutenberg uses it and aligning with them seems good if others are okay with it. I find space aligned text annoying because I use a proportional font and they look very wonky and bad to me, but for most folks it would be easier to read.
        "jsdoc/check-param-names": ['error'], // Keeps the documentation up-to-date with implementation
        "jsdoc/check-property-names": ['error'], // Hopefully this one becomes irrelevant as we move to TypeScript but doesn't hurt
        "jsdoc/check-syntax": ['off'], TypeScript compiler will catch this when it actually matters
        "jsdoc/check-tag-names": ['error'], // With the additional `@values` tag you added for Vue
        "jsdoc/check-types": ['off'], // Again, TS will catch this for us
        "jsdoc/check-values": ['off'], // This doesn't check anything that would matter to us
        "jsdoc/empty-tags": ['error'], // Ensure tags that aren't meant to have values don't have them
        "jsdoc/implements-on-classes": ['off'], // Avoid using JSDoc as a replacement for class-based semantics that don't really exist in JavaScript anyway
        "jsdoc/match-description": ['error'], // Sure why not, we do this in Python anyway.
        "jsdoc/multiline-blocks": ['error'], // Prevents wonky formatting
        "jsdoc/newline-after-description": ['error'], // More whitespace is good
        "jsdoc/no-bad-blocks": ['error'], // Keeps two asterisks on the opening line
        "jsdoc/no-defaults": ['off'], // Allows us to document default values
        "jsdoc/no-missing-syntax": ['off'], // I'm not sure how this one works, I don't think we need it
        "jsdoc/no-multi-asterisks": ['error', { allowWhitespace: true }], // Seems sensible to me
        "jsdoc/no-restricted-syntax": ['off'], // I don't think we need this and we'd have to configure specifics for it anyway
        "jsdoc/no-types": ['off'], // I guess because we're using the TSDoc linter for TS we don't need this
        "jsdoc/no-undefined-types": ['off'], // This rule is broken with TypeScript type params
        "jsdoc/require-asterisk-prefix": ['error'], // No comment
        "jsdoc/require-description": ['off'],
        "jsdoc/require-description-complete-sentence": ['error'], // Matches our Python style (I think)
        "jsdoc/require-example": ['off'],
        "jsdoc/require-file-overview": ['off'],
        "jsdoc/require-hyphen-before-param-description": ['error', "never"], // My own preference is to not use the hyphen, I don't think we use it currently, but feel free to change this to 'always' if that's what folks want
        "jsdoc/require-jsdoc": ['off'], // Not sure we need this
        "jsdoc/require-param": ['error'],
        "jsdoc/require-param-description": ['error'],
        "jsdoc/require-param-name": ['error'],
        "jsdoc/require-param-type": ['error'], // TS might catch this too, not sure, but it might also just use implicit any which we don't want. Might as well enforce adding types.
        "jsdoc/require-property": ['errpr'], // Prevent `object` and `any` aliases
        "jsdoc/require-property-description": ['error'],
        "jsdoc/require-property-name": ['error'],
        "jsdoc/require-property-type": ['error'],
        "jsdoc/require-returns": ['error'],
        "jsdoc/require-returns-check": ['error'], // This one seems sensible, prevents adding a `return` annotation when the function doesn't return anything
        "jsdoc/require-returns-description": ['error'],
        "jsdoc/require-returns-type": ['error'],
        "jsdoc/require-throws": ['off'], // We're not writing Java :slightly_smiling_face: 
        "jsdoc/require-yields": ['error'], // Don't think we have any generators but might as well document them
        "jsdoc/require-yields-check": ['error'],
        "jsdoc/tag-lines": ['error'], // Stop adding newlines between tags... Not sure about this one TBH
        "jsdoc/valid-types": ['off'] // TS will catch this when it matters
    }
```